ethereum/core/src/main/java/org/hyperledger/besu/ethereum/worldstate/Pruner.java

```diff
@@ -84,12 +84,14 @@ public Pruner(
   }
 
   public void start() {
-
     if (state.compareAndSet(State.IDLE, State.RUNNING)) {
```

Comment:
I think we could call this as many times as we like but the `compareAndSet` would make sure that the if body is only executed once, right? That said, I don't mind pushing the whole thing into the execute. Works for me.

Reply:
The executor may back up, so we could queue up a bunch of the lambdas for execution before the first one starts (which is where I presume the stat gets set to running or prepare).  While the test is idempotent the execution does not appear on its surface to be, especially since we are adding observers inside the async function.

Reply:
The `state` variable  is an `AtomicReference` - the code inside the block is guaranteed to execute at most once as it's currently written.

Reply:
nm - I think I see the issue:  the startup logic could actually run after the shutdown logic because you don't know when it will actually execute 👍